lib/src/parsers/type-parser.ts

```diff
@@ -455,6 +461,43 @@ function parseUnionType(
   }
 }
 
+/**
+ * Parse an intersection type node.
+ *
+ * @param typeNode intersection type node
+ * @param typeTable a TypeTable
+ * @param lociTable a LociTable
+ */
+function parseIntersectionTypeNode(
+  typeNode: IntersectionTypeNode,
+  typeTable: TypeTable,
+  lociTable: LociTable
+): Result<Type, ParserError> {
+  const allowedTargetTypes = typeNode
+    .getTypeNodes()
+    .filter(type => !type.getType().isUndefined());
+  const types: Type[] = [];
+  for (const tn of allowedTargetTypes) {
+    const typeResult = parseType(tn, typeTable, lociTable);
+    if (typeResult.isErr()) return typeResult;
+    // Only allow objects, unions, intersections and references
+    const typeResultType = typeResult.unwrap();
+    if (isPrimitiveType(typeResultType) || isArrayType(typeResultType)) {
```

Comment:
Resolve/dereference to concrete types and check if the resulting type is an object type